core/src/main/java/io/syndesis/core/Names.java

```diff
@@ -39,13 +39,22 @@ private Names() {
      */
     public static String sanitize(String name) {
         String trimmed = name.length() > MAXIMUM_NAME_LENGTH ? name.substring(0, MAXIMUM_NAME_LENGTH) : name;
+        final int[] last = new int[]{-1};
+
         return trimmed
             .replaceAll(SPACE, DASH)
             .replaceAll(INVALID_CHARACTER_REGEX, BLANK)
             .toLowerCase(Locale.US)
             .chars()
             .filter(i -> !String.valueOf(i).matches(INVALID_CHARACTER_REGEX))
-            .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
+            .collect(StringBuilder::new,
+                (b, i) -> {
+                    if (i == '-' && last[0] == '-') {
```

Comment:
Actually the trick to update an outer array feels a bit dirty ;-)

maybe an anonymous class with a member variable instead of a lambda would be a better fit ?